tests/test_utils.py

```diff
@@ -30,6 +33,50 @@ def test_not_exclusive(self) -> None:
         self.assertEqual(nodes.are_exclusive(xass1, xnames[1]), False)
         self.assertEqual(nodes.are_exclusive(xass1, xnames[2]), False)
 
+    @pytest.mark.skipif(not PY38_PLUS, reason="needs assignment expressions")
+    def test_not_exclusive_walrus_operator(self) -> None:
+        from astroid import extract_node, nodes
+
+        node_if, node_body, node_or_else = extract_node(
+            """
+        if val := True:  #@
```

Comment:
I would also like a test for `val := True or val_2 := True` as then `val_2` isn't necessarily assigned.

Reply:
I understand you. I was also thinking about that test. But then I realized that this bug is about `are_exclusive` function and this test is about `are_inclusive`.

## `are_exclusive` logic
When I saw `are_exclusive` docstring the first time (`return true if the two given statements are mutually exclusive`), I was wondering what `exclusive` mean here. 
Correct me if I am wrong, but in `are_exclusive` we test:
- if one statement (`stmt1`) is executed, the other one (`stmt2`) is not executed,
- AND if the other statement (`stmt2`) is executed, then the first statement (`stmt1`) is not executed.

## Exclusive for `or` expression
Analogous code to your `or` example, but without walrus operator:
```
val, val_2 = extract_node(
    """
val = True  #@
if val:
    print('ok')
else:
    val_2 = True  #@
    if val_2:
        print('ok')
    else:
        print('not')
"""
)
assert nodes.are_exclusive(val_1, val_2) is False
```
Both version (before this PR, after this PR) have same result (`False`), which I believe is correct.


## Missing warning/error in Pylint
Unfortunately, for this code, I see no warning in `pylint`:
```
if (val := True) or (val_2 := True):  #@
    print(val)  #@
    print(val_2)  #@
else:
    print(val)  #@
    print(val_2)  #@
```
But of course, it is `NameError` in runtime (`NameError: name 'val_2' is not defined`). 
PyCharm shows me here a warning: `Name 'val_2' can be undefined`.


## Inclusive
My understanding of `inclusive`:  `stmt1` is in "inclusive" relation with `stmt2` if `stmt1` is __always__ executed before `stmt2`.
```
if a or b:
    print(c)
else:
    print(d)
```
Possible relations between 2 statements:
- exclusive `(c,d)`, `(d,c)`,
- inclusive `(a,c)`, `(a,d)` - note missing `(b,c)` - we are unsure whenever `b` will be executed.

Therefore, if `are_exclusive` function return `False`, we cannot assume easy that these statements are inclusive.
I think maybe there should exist a similar function `are_inclusive`, because I see in the code few places where `are_exclusive` is used with negation (which I believe, as I show above, it may be incorrect in some cases).

Reply:
Thanks for this clear explanation. Your follow-up seems logical to me, but indeed for now this is enough.

Perhaps we should convert your investigation into a new issue?

Reply:
> Perhaps we should convert your investigation into a new issue?

Yes, it is clearly another issue.

> for now this is enough.

Nonetheless, I added that test to be sure what is expected result for this function.